### PR TITLE
fix(egw): endpoint gateway access pending

### PR DESCRIPTION
**Pull request by deepaksibm**
Fix description. When the EGW is created for a PPSG and the PPSG has the access policy of "review", then the EGW create should not taint on timeout, as 'pending' is a valid state for EGW when PPSG is in review mode.  User can either perform a permit or deny action at a later point in time as well which will make the EGW to either go to stable or failed state depending on the action

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go [hunk 1]
```diff
@@ -385,6 +386,26 @@ func resourceIBMisVirtualEndpointGatewayCreate(d *schema.ResourceData, meta inte
 	if d.Get(targetResourceTypeFmt).(string) == "private_path_service_gateway" {
 		_, err = isWaitForVirtualEndpointGatewayForPPSGAvailable(sess, d.Id(), d.Timeout(schema.TimeoutCreate))
 		if err != nil {
+			if err != nil {
```
Comment (ujjwal-ibm) [inline ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go:389]: err is not nil here already

**Review by ujjwal-ibm — CHANGES_REQUESTED** [review]

**Comment by deepaksibm** [comment]
```
$ tfpt TestAccIBMISVirtualEndpointGateway_Basic
=== RUN   TestAccIBMISVirtualEndpointGateway_Basic
--- PASS: TestAccIBMISVirtualEndpointGateway_Basic (127.32s)
=== RUN   TestAccIBMISVirtualEndpointGateway_Basic_SecurityGroups
--- PASS: TestAccIBMISVirtualEndpointGateway_Basic_SecurityGroups (127.51s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     258.040s


$ tfpt TestAccIBMISVirtualEndpointGateway_PPSG_With_AccessPolicy_Review_And_Timeout
=== RUN   TestAccIBMISVirtualEndpointGateway_PPSG_With_AccessPolicy_Review_And_Timeout
--- PASS: TestAccIBMISVirtualEndpointGateway_PPSG_With_AccessPolicy_Review_And_Timeout (1997.14s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     2000.278s
```

File: ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go (hunk 1 above)
Comment (deepaksibm) [reply ibm/service/vpc/resource_ibm_is_virtual_endpoint_gateway.go]: updated the logic

**Comment by deepaksibm** [comment]
Upstream PR merged

**Pull request closed without merging**